Content.Client/Traits/ParacusiaSystem.cs

```diff
@@ -0,0 +1,66 @@
+using Content.Shared.Audio;
+using Content.Shared.Traits.Assorted;
+using Content.Client.Camera;
+using Robust.Shared.Random;
+using Robust.Shared.Player;
+using Robust.Shared.Map;
+
+namespace Content.Client.Traits;
+
+public sealed class ParacusiaSystem : EntitySystem
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly CameraRecoilSystem _camera = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<ParacusiaComponent, ComponentStartup>(SetupParacusia);
+    }
+
+    private void SetupParacusia(EntityUid uid, ParacusiaComponent component, ComponentStartup args)
+    {
+        component.NextIncidentTime =
+            _random.NextFloat(component.minTimeBetweenIncidents, component.maxTimeBetweenIncidents);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        foreach (var paracusia in EntityQuery<ParacusiaComponent>())
+        {
+            paracusia.NextIncidentTime -= frameTime;
+
+            if (paracusia.NextIncidentTime >= 0)
+                continue;
+
+            // Set the new time.
+            paracusia.NextIncidentTime +=
+                _random.NextFloat(paracusia.minTimeBetweenIncidents, paracusia.maxTimeBetweenIncidents);
+
+            List<string> sounds = paracusia.Sounds ?? new List<string> { };
+            if (paracusia.Sounds == null || paracusia.Sounds.Count == 0)
+                return;
```

Comment:
This is still returning in a loop, though you should just check if the attached entity has paracusia anyway.

The accumulator also needs to be a timespan.